packages/amazonq/test/unit/codewhisperer/util/codeParsingUtil.test.ts

```diff
@@ -57,6 +60,90 @@ describe('RegexValidationForJava', () => {
 })
 
 describe('isTestFile', () => {
+    beforeEach(async function () {
+        tempFolder = (await createTestWorkspaceFolder()).uri.fsPath
+    })
+
+    it('validate by file path', async function () {
+        const langs = new Map<string, string>([
+            ['java', '.java'],
+            ['python', '.py'],
+            ['typescript', '.ts'],
+            ['javascript', '.js'],
+            ['typescriptreact', '.tsx'],
+            ['javascriptreact', '.jsx'],
+        ])
+        const testFilePathsWithoutExt = [
+            '/test/MyClass',
+            '/test/my_class',
+            '/tst/MyClass',
+            '/tst/my_class',
+            '/tests/MyClass',
+            '/tests/my_class',
+        ]
+
+        const srcFilePathsWithoutExt = [
+            '/src/MyClass',
+            'MyClass',
+            'foo/bar/MyClass',
+            'foo/my_class',
+            'my_class',
+            'anyFolderOtherThanTest/foo/myClass',
+        ]
+
+        for (const [languageId, ext] of langs) {
+            const testFilePaths = testFilePathsWithoutExt.map((it) => it + ext)
+            for (const testFilePath of testFilePaths) {
+                const actual = await isTestFile(testFilePath, { languageId: languageId })
+                assert.strictEqual(actual, true)
+            }
+
+            const srcFilePaths = srcFilePathsWithoutExt.map((it) => it + ext)
+            for (const srcFilePath of srcFilePaths) {
+                const actual = await isTestFile(srcFilePath, { languageId: languageId })
+                assert.strictEqual(actual, false)
+            }
+        }
+    })
+
+    async function assertIsTestFile(srcFiles: string[], tstFiles: string[], fileExt: string) {
+        for (const name of tstFiles) {
+            const file = `${name}.${fileExt}`
+            const editor = await openATextEditorWithText('', file, tempFolder, { preview: false })
+            const actual = await isTestFile(editor.document.uri.fsPath, { languageId: editor.document.languageId })
+            assert.strictEqual(actual, true)
+        }
+
+        for (const name of srcFiles) {
+            const file = `${name}.${fileExt}`
+            const editor = await openATextEditorWithText('', file, tempFolder, { preview: false })
+            const actual = await isTestFile(editor.document.uri.fsPath, { languageId: editor.document.languageId })
+            assert.strictEqual(actual, false)
+        }
+    }
+
+    it('validate by file name', async function () {
```

Comment:
if i'm reading the logs right, this test takes 10 seconds. that's not acceptable.

`openATextEditorWithText` also opens an editor, which can be slow. is that actually necessary, or could we add a param to it which only calls `vscode.workspace.openTextDocument` and skips `showTextDocument` (assuming I'm correct about the source of the slowness)